src/Console/Command/ProcessCommand.php

```diff
@@ -169,12 +179,15 @@ private function processFile(SplFileInfo $fileInfo, int &$i): void
             $newContent = $this->fileProcessor->processFileToString($fileInfo);
 
             if ($newContent !== $oldContent) {
-                $this->symfonyStyle->writeln(sprintf('<options=bold>%d) %s</>', $i, $fileInfo->getPathname()));
-                $this->symfonyStyle->writeln($this->differAndFormatter->diffAndFormat($oldContent, $newContent));
+                $this->consoleStyle->writeln(sprintf('<options=bold>%d) %s</>', $i, $fileInfo->getPathname()));
+                $this->consoleStyle->writeln($this->differAndFormatter->diffAndFormat($oldContent, $newContent));
                 ++$i;
             }
         } else {
-            $this->fileProcessor->processFile($fileInfo);
+            $newContent = $this->fileProcessor->processFile($fileInfo);
+            if ($newContent !== false) {
+                $this->changedFiles[] = $fileInfo->getPathname();
```

Comment:
Could use immidiate report as in the code above?

```php
$this->consoleStyle->writeln(sprintf('<options=bold>%d) %s</>', $i, $fileInfo->getPathname()));
```

It is preffereable to see any output right away than wait till the end to see all changed files. So people can see the Rector is doing some real work

It would also simplify rest of code

Reply:
That a good idea actually. In case `Rector` fails, it had already shown what have already changed :+1:

Reply:
As we are updating the progress bar together with the process of `processFile`, we'll need to figure out a way to do not conflict the two commands:
![captura de tela de 2017-12-28 15-19-15](https://user-images.githubusercontent.com/16328050/34417989-abf63e3a-ebe2-11e7-8039-fc018be2b585.png)

Reply:
Ah, I had a scratch in the back of my head it could be the problem.

In that case it would be the best to do it like you suggest - collect first, than render second. Even for the code above.

Like PHPUnit - progress bar first, result second. Maybe that's the reason why it is done this way :sweat_smile: